cmd/k8s-operator/proxygroup.go

```diff
@@ -138,7 +139,28 @@ func (r *ProxyGroupReconciler) Reconcile(ctx context.Context, req reconcile.Requ
 		return setStatusReady(pg, metav1.ConditionFalse, reasonProxyGroupInvalid, message)
 	}
 
-	if err = r.maybeProvision(ctx, pg); err != nil {
+	proxyClassName := r.defaultProxyClass
+	if pg.Spec.ProxyClass != "" {
+		proxyClassName = pg.Spec.ProxyClass
+	}
+
+	var proxyClass *tsapi.ProxyClass
+	if proxyClassName != "" {
+		proxyClass = new(tsapi.ProxyClass)
+		if err := r.Get(ctx, types.NamespacedName{Name: proxyClassName}, proxyClass); err != nil {
+			logger.Errorf("error getting ProxyGroup's ProxyClass %s: %w", proxyClassName, err)
```

Comment:
Nit: we don't need to log this line- c/r will log the same as we are returning it as an error from Reconcile, so this would appear twice

Reply:
Thanks for spotting this. There were a few other occurrences in this function, I fixed those up too.